harness/determined/deploy/aws/aws.py

```diff
@@ -62,8 +66,23 @@ def stop_master(master_id: str, boto3_session: boto3.session.Session, delete: bo
 def delete(stack_name: str, boto3_session: boto3.session.Session) -> None:
     ec2 = boto3_session.client("ec2")
 
+    # Check if we have stack outputs containing ec2 instance and
+    # bucket information, if not, just remove the stack.
+    try:
+        stack_output = get_output(stack_name, boto3_session)
+    except NoStackOutputError:
+        print(
+            f"Stack {stack_name} is in inconsistent state. "
+            "This error is ignored as stack is going to be deleted."
+        )
+        if stack_uses_spot(stack_name, boto3_session):
+            print("Terminating Running Agents and Pending Spot Requests")
+            clean_up_spot(stack_name, boto3_session)
```

Comment:
it seems wrong that `clean_up_spot` doesn't need stack outputs, and reconstructs spot instance tags by itself (instead of using the values from output like `terminate_running_agents`).
so I think we should either
1. remove this block, lines 78-81, because it's unlikely there are any spot instances anyway,
2. or preface this block with a comment why we only delete spot instances (and not normal ones or s3 bucket).

Reply:
Sounds good. I didn't really like that block being in there twice anyway.

Reply:
lgtm, feel free to land when the checks pass